src/backend/distributed/operations/worker_node_manager.c

```diff
@@ -584,15 +584,30 @@ WorkerNodeCompare(const void *lhsKey, const void *rhsKey, Size keySize)
 	const WorkerNode *workerLhs = (const WorkerNode *) lhsKey;
 	const WorkerNode *workerRhs = (const WorkerNode *) rhsKey;
 
+	return NodeNamePortCompare(workerLhs->workerName, workerRhs->workerName,
+							   workerLhs->workerPort, workerRhs->workerPort);
+}
+
 
-	int nameCompare = strncmp(workerLhs->workerName, workerRhs->workerName,
-							  WORKER_LENGTH);
+/*
+ * NodeNamePortCompare implements the common logic for comparing two nodes
+ * with their given nodeNames and ports.
+ *
+ * This function is useful for ensuring consistency of sort operations between
```

Comment:
Minor: Do we need this information here? 
I am asking because the function itself makes sense that it is used for comparing nodename/port pairs.

Reply:
Just in case someone decides to change the ordering in any of these functions, they might see this comment and re-think :)